google-cloud-bigquery/lib/google/cloud/bigquery/table.rb

```diff
@@ -2010,8 +2014,8 @@ def load_local file, options = {}
 
         def storage_url? file
           file.respond_to?(:to_gs_url) ||
-            (file.respond_to?(:to_str) &&
-            file.to_str.downcase.start_with?("gs://"))
+            (file.respond_to?(:to_s) &&
```

Comment:
This is not a good change. The existence of a `to_s` method is not an indication that the object is a string, and nearly all objects have this method. Non-string objects such as File, Integer, Float, `nil`, etc. all respond to `to_s`.

Reply:
Understood. This was the easier approach; I didn't know which direction you wanted to go. To me the "ruby" approach would be that anything that "acts like a" gs URL string is acceptable. 

`File`, `Integer`, `Float`, and `nil` would all fail the second half of the check:

```
file.to_s.downcase.start_with?("gs://"))
```

Happy to make the change for more explicit `URI`  support, just suggesting a less constrictive option.

Reply:
That's a fair point. I'll leave it to @quartzmo to decide.

Reply:
I'm OK with any approach that works. More explicit `URI` support seems fine.

Reply:
The current approach works. Would you like me to change it to be more explicit/restrictive about `URI`? You're call.

Reply:
Yes, if you don't mind, let's go with the explicit URI support ("a conditional to the beginning of the method that checks if the object is a URI and convert it to a String." as @blowmage suggested). Thanks @jeremywadsack, appreciate your time and work on this!